applications/luci-app-filemanager/htdocs/luci-static/resources/view/system/filemanager.js

```diff
@@ -0,0 +1,4355 @@
+
+		console.log(`scrollToMatch called with index: ${index}`);
+		console.log(`lineNumber: ${lineNumber}`);
+		console.log(`newScrollTop: ${newScrollTop}`);
+
+		// Update the scrollTop property to trigger handleScroll
+		this.hexeditContent.scrollTop = newScrollTop;
+	}
+
+	/**
+	 * Clears previous search highlights.
+	 */
+	clearSearchHighlights() {
+		// Remove previous highlights
+		this.hexview.querySelectorAll('.search-highlight').forEach(span => {
+			span.classList.remove('search-highlight');
+		});
+		this.textview.querySelectorAll('.search-highlight').forEach(span => {
+			span.classList.remove('search-highlight');
+		});
+
+		// Reset active view
+		this.activeView = null;
+
+		// Reset all search status fields to 0/0
+		['ascii', 'hex', 'regex'].forEach(type => {
+			this.updateSearchStatus(type, 0, 0);
+		});
+	}
+
+	/**
+	 * Calculates the number of visible rows based on the container's height.
+	 */
+	calculateVisibleRows() {
+		const lineHeight = 16; // Height of one row in pixels
+		const containerHeight = this.hexeditContent.clientHeight;
+		this.visibleRows = Math.floor(containerHeight / lineHeight);
+		this.visibleByteCount = this.bytesPerRow * this.visibleRows;
+		console.log(`calculateVisibleRows: visibleRows=${this.visibleRows}, visibleByteCount=${this.visibleByteCount}`);
+		this.renderDom(); // Re-render to apply the new rows
+	}
+
+	/**
+	 * Sets the data to be displayed in the hex editor.
+	 *
+	 * @param {Uint8Array} data - The data to set.
+	 */
+	setData(data) {
+		this.data = data;
+		this.totalRows = Math.ceil(this.data.length / this.bytesPerRow);
+		console.log(`setData: data length=${this.data.length}, totalRows=${this.totalRows}`);
+		this.calculateVisibleRows(); // Ensure visibleRows are calculated before rendering
+	}
+
+	/**
+	 * Retrieves the current data from the hex editor.
+	 *
+	 * @returns {Uint8Array} - The current data.
+	 */
+	getData() {
+		return this.data;
+	}
+
+	/**
+	 * Handles the scroll event for virtual scrolling.
+	 *
+	 * @param {Event} event - The scroll event.
+	 */
+	handleScroll(event) {
+		const scrollTop = this.hexeditContent.scrollTop;
+		const lineHeight = 16; // Approximate height of a byte row in pixels
+		const firstVisibleLine = Math.floor(scrollTop / lineHeight);
+		const newStartIndex = firstVisibleLine * this.bytesPerRow;
+
+		console.log(`handleScroll: scrollTop=${scrollTop}, firstVisibleLine=${firstVisibleLine}, newStartIndex=${newStartIndex}`);
+
+		// Update startIndex and re-render the DOM if necessary
+		if (newStartIndex !== this.startIndex) {
+			this.startIndex = newStartIndex;
+			this.renderDom(); // Re-render visible data
+			console.log(`handleScroll: Updated startIndex and rendered DOM.`);
+		}
+	}
+
+	/**
+	 * Renders the visible portion of the hex editor based on the current scroll position.
+	 */
+	renderDom() {
+		// Clear existing content
+		this.offsets.innerHTML = '';
+		this.hexview.innerHTML = '';
+		this.textview.innerHTML = '';
```

Comment:
perhaps something like this to avoid triplicate:
```js
[this.offsets, this.hexview, this.textview].forEach(view => view.innerHTML = '');
```

it's only slightly more compact - but it cleans up the rest of the code a bit. Your call.

Reply:
Done